pkg/loki/config_wrapper.go

```diff
@@ -116,6 +116,27 @@ func (c *ConfigWrapper) ApplyDynamicConfig() cfg.Source {
 	}
 }
 
+func lastConfigFor(configs []config.PeriodConfig, predicate func(config.PeriodConfig) bool) int {
+	for i := len(configs) - 1; i >= 0; i-- {
+		if predicate(configs[i]) {
+			return i
+		}
+	}
+	return len(configs)
```

Comment:
same question as above, but asked closer to the source:
What was the reason for using `len(r.SchemaConfig.Configs` as your sentinel value when a config can't be found here? I wonder if `-1` might be a more canonical?

Reply:
I see the motivation for providing the last config when a more specific config of the correct type can't be found, but since we're only using this to find configs the apply better BoltdbShipper or TSDBConfig defaults for, I don't think we need to care about using the last config for anything.

Reply:
your comment above explained it, following the convention from `sort.Search` :+1: